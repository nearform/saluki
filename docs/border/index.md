# Border

---

## Configuration

Configurable via;

```
border
```
