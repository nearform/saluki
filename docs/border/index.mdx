# Border

---

## Rules

```js
border.color.black
border.color.white

border.color.lighterGray
border.color.lightGray
border.color.gray
border.color.darkGray
border.color.darkerGray

border.color.lighterRed
border.color.lightRed
border.color.red
border.color.darkRed
border.color.darkerRed

border.color.lighterBlue
border.color.lightBlue
border.color.blue
border.color.darkBlue
border.color.darkerBlue

border.color.lighterYellow
border.color.lightYellow
border.color.yellow
border.color.darkYellow
border.color.darkerYellow

border.color.lighterGreen
border.color.lightGreen
border.color.green
border.color.darkGreen
border.color.darkerGreen

border.color.lighterOrange
border.color.lightOrange
border.color.orange
border.color.darkOrange
border.color.darkerOrange

border.color.lighterPurple
border.color.lightPurple
border.color.purple
border.color.darkPurple
border.color.darkerPurple

border.color.top.black
border.color.top.white

border.color.top.lighterGray
border.color.top.lightGray
border.color.top.gray
border.color.top.darkGray
border.color.top.darkerGray

border.color.top.lighterRed
border.color.top.lightRed
border.color.top.red
border.color.top.darkRed
border.color.top.darkerRed

border.color.top.lighterBlue
border.color.top.lightBlue
border.color.top.blue
border.color.top.darkBlue
border.color.top.darkerBlue

border.color.top.lighterYellow
border.color.top.lightYellow
border.color.top.yellow
border.color.top.darkYellow
border.color.top.darkerYellow

border.color.top.lighterGreen
border.color.top.lightGreen
border.color.top.green
border.color.top.darkGreen
border.color.top.darkerGreen

border.color.top.lighterOrange
border.color.top.lightOrange
border.color.top.orange
border.color.top.darkOrange
border.color.top.darkerOrange

border.color.top.lighterPurple
border.color.top.lightPurple
border.color.top.purple
border.color.top.darkPurple
border.color.top.darkerPurple

border.color.right.black
border.color.right.white

border.color.right.lighterGray
border.color.right.lightGray
border.color.right.gray
border.color.right.darkGray
border.color.right.darkerGray

border.color.right.lighterRed
border.color.right.lightRed
border.color.right.red
border.color.right.darkRed
border.color.right.darkerRed

border.color.right.lighterBlue
border.color.right.lightBlue
border.color.right.blue
border.color.right.darkBlue
border.color.right.darkerBlue

border.color.right.lighterYellow
border.color.right.lightYellow
border.color.right.yellow
border.color.right.darkYellow
border.color.right.darkerYellow

border.color.right.lighterGreen
border.color.right.lightGreen
border.color.right.green
border.color.right.darkGreen
border.color.right.darkerGreen

border.color.right.lighterOrange
border.color.right.lightOrange
border.color.right.orange
border.color.right.darkOrange
border.color.right.darkerOrange

border.color.right.lighterPurple
border.color.right.lightPurple
border.color.right.purple
border.color.right.darkPurple
border.color.right.darkerPurple

border.color.bottom.black
border.color.bottom.white

border.color.bottom.lighterGray
border.color.bottom.lightGray
border.color.bottom.gray
border.color.bottom.darkGray
border.color.bottom.darkerGray

border.color.bottom.lighterRed
border.color.bottom.lightRed
border.color.bottom.red
border.color.bottom.darkRed
border.color.bottom.darkerRed

border.color.bottom.lighterBlue
border.color.bottom.lightBlue
border.color.bottom.blue
border.color.bottom.darkBlue
border.color.bottom.darkerBlue

border.color.bottom.lighterYellow
border.color.bottom.lightYellow
border.color.bottom.yellow
border.color.bottom.darkYellow
border.color.bottom.darkerYellow

border.color.bottom.lighterGreen
border.color.bottom.lightGreen
border.color.bottom.green
border.color.bottom.darkGreen
border.color.bottom.darkerGreen

border.color.bottom.lighterOrange
border.color.bottom.lightOrange
border.color.bottom.orange
border.color.bottom.darkOrange
border.color.bottom.darkerOrange

border.color.bottom.lighterPurple
border.color.bottom.lightPurple
border.color.bottom.purple
border.color.bottom.darkPurple
border.color.bottom.darkerPurple

border.color.left.black
border.color.left.white

border.color.left.lighterGray
border.color.left.lightGray
border.color.left.gray
border.color.left.darkGray
border.color.left.darkerGray

border.color.left.lighterRed
border.color.left.lightRed
border.color.left.red
border.color.left.darkRed
border.color.left.darkerRed

border.color.left.lighterBlue
border.color.left.lightBlue
border.color.left.blue
border.color.left.darkBlue
border.color.left.darkerBlue

border.color.left.lighterYellow
border.color.left.lightYellow
border.color.left.yellow
border.color.left.darkYellow
border.color.left.darkerYellow

border.color.left.lighterGreen
border.color.left.lightGreen
border.color.left.green
border.color.left.darkGreen
border.color.left.darkerGreen

border.color.left.lighterOrange
border.color.left.lightOrange
border.color.left.orange
border.color.left.darkOrange
border.color.left.darkerOrange

border.color.left.lighterPurple
border.color.left.lightPurple
border.color.left.purple
border.color.left.darkPurple
border.color.left.darkerPurple

border.width.none
border.width.thin
border.width.medium
border.width.thick
border.width.extraThick

border.width.top.none
border.width.top.thin
border.width.top.medium
border.width.top.thick
border.width.top.extraThick

border.width.right.none
border.width.right.thin
border.width.right.medium
border.width.right.thick
border.width.right.extraThick

border.width.bottom.none
border.width.bottom.thin
border.width.bottom.medium
border.width.bottom.thick
border.width.bottom.extraThick

border.width.left.none
border.width.left.thin
border.width.left.medium
border.width.left.thick
border.width.left.extraThick

border.style.solid
border.style.dashed
border.style.dotted
border.style.none

border.style.top.solid
border.style.top.dashed
border.style.top.dotted
border.style.top.none

border.style.right.solid
border.style.right.dashed
border.style.right.dotted
border.style.right.none

border.style.bottom.solid
border.style.bottom.dashed
border.style.bottom.dotted
border.style.bottom.none

border.style.left.solid
border.style.left.dashed
border.style.left.dotted
border.style.left.none

border.radius.small
border.radius.medium
border.radius.large

border.radius.topRight.small
border.radius.topRight.medium
border.radius.topRight.large

border.radius.bottomRight.small
border.radius.bottomRight.medium
border.radius.bottomRight.large

border.radius.bottomLeft.small
border.radius.bottomLeft.medium
border.radius.bottomLeft.large

border.radius.topLeft.small
border.radius.topLeft.medium
border.radius.topLeft.large
```

## Configuration

Configurable via;

```
border
```

## Defaults

```js
{
  width: {
    none: 'none',
    thin: '1px',
    medium: '2px',
    thick: '4px',
    extraThick: '8px'
  },
  radius: {
    small: '2px',
    medium: '4px',
    large: '6px'
  }
}
```
