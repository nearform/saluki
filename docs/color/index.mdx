import { Color } from './'

# Color

---

## Example

<Color>Hello</Color>

```js
const Color = styled.h1`
  ${color.white}
  ${border.solid}
  ${border.blue}
  ${background.black}
`
```

## Configuration

Configurable via;

```
color
```

## Defaults

```js
{
  black: 'black',
  blue: 'blue',
  gray: 'gray',
  green: 'green',
  orange: 'orange',
  red: 'red',
  purple: 'purple',
  yellow: 'yellow',
  white: 'white'
}
```
