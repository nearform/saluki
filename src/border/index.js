const borderWidths = { none: 0, thin: 1, medium: 2, thick: 4, extraThick: 8 }
const borderStyles = {
  solid: 'solid',
  dashed: 'dashed',
  dotted: 'dotted',
  none: 'none'
}

const setupBorderColors = (border, colors) => {
  for (const [key, value] of Object.entries(colors)) {
    border[key] = {
      'border-color': value
    }
    border['top'][key] = {
      'border-top-color': value
    }
    border['right'][key] = {
      'border-right-color': value
    }
    border['bottom'][key] = {
      'border-bottom-color': value
    }
    border['left'][key] = {
      'border-left-color': value
    }
  }
}

const setupBorderProperties = (border, values, name, valTransform) => {
  for (const [key, value] of Object.entries(values)) {
    const val = valTransform ? valTransform(value) : value
    border[key] = {
      [`border-${name}`]: val
    }
    border['top'][key] = { [`border-top-${name}`]: val }
    border['right'][key] = { [`border-right-${name}`]: val }
    border['bottom'][key] = { [`border-bottom-${name}`]: val }
    border['left'][key] = { [`border-left-${name}`]: val }
  }
}
const setupBorder = (customWidths = {}, colors = {}) => {
  let border = {
    border: 'solid 1px black',
    top: { ...borderStyles },
    right: { ...borderStyles },
    bottom: { ...borderStyles },
    left: { ...borderStyles }
  }
  setupBorderColors(border, colors)
  setupBorderProperties(
    border,
    { ...borderWidths, ...customWidths },
    'width',
    v => `${v}px`
  )
  setupBorderProperties(border, borderStyles, 'style')
  return border
}

export default setupBorder
