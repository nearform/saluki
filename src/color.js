export const color = {
  red: 'red'
}
