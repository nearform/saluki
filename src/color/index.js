export const defaultRules = {
  black: 'black',
  white: 'white',
  lighterGray: '#d5d9dc',
  lightGray: '#9ea6ae',
  gray: '#495057',
  darkGray: '#3a4045',
  darkerGray: '#23262a',
  lighterRed: '#f9b8b8',
  lightRed: '#f25959',
  red: '#f03e3e',
  darkRed: '#a60d0d',
  darkerRed: '#470606',
  lighterBlue: '#bbdaf6',
  lightBlue: '#61a9ea',
  blue: '#1c7ed6',
  darkBlue: '#155d9e',
  darkerBlue: '#092844',
  lighterYellow: '#ffe4b3',
  lightYellow: '#ffc04d',
  yellow: '#f59f00',
  darkYellow: '#b37400',
  darkerYellow: '#4d3200',
  lighterGreen: '#c5edcc',
  lightGreen: '#77d587',
  green: '#37b24d',
  darkGreen: '#2a883b',
  darkerGreen: '#123a19',
  lighterOrange: '#fdd2b5',
  lightOrange: '#fa9551',
  orange: '#f76707',
  darkOrange: '#ae4805',
  darkerOrange: '#4a1f02',
  lighterPurple: '#e6c3ee',
  lightPurple: '#c474d8',
  purple: '#ae3ec9',
  darkPurple: '#78278b',
  darkerPurple: '#33113c'
}

const setupColor = colors => {
  let color = {}
  for (const [key, value] of Object.entries(colors)) {
    color[key] = {
      color: value
    }
  }
  return color
}

export default setupColor
