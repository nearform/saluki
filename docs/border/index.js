import styled from 'styled-components/macro'
import { border } from '../../src'

export const Color = styled.h1`
  ${border.solid}
  ${border.blue}
`

export const Style = styled.h1`
  ${border.bottom.dashed}
`

export const Width = styled.h1`
  ${border.solid}
  ${border.thick}
`
