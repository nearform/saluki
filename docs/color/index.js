import styled from 'styled-components/macro'
import { background, border, color } from '../../src'

export const Color = styled.h1`
  ${color.white}
  ${border.solid}
  ${border.blue}
  ${background.black}
`
