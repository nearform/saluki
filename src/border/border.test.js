import border, { defaultRules, combineRules } from './'
import { color } from '../color'

describe('border', () => {
  describe('when no custom config is provided', () => {
    it('should return the default config', () => {
      const combinedRules = combineRules(
        defaultRules,
        {
          width: {},
          radius: {}
        },
        color
      )
      expect(border(combinedRules)).toMatchSnapshot()
    })
  })

  describe('when custom config is provided', () => {
    describe('when the custom config overrides a value', () => {
      it('should replace the default value', () => {
        const combinedRules = combineRules(
          defaultRules,
          {
            width: { thin: '.5rem' }
          },
          color
        )
        expect(border(combinedRules)).toMatchSnapshot()
      })
    })

    describe('when the custom config provides a new value', () => {
      it('should add the value and not amend any existing values', () => {
        const combinedRules = combineRules(
          defaultRules,
          {
            radius: { huge: '20px' }
          },
          color
        )
        expect(border(combinedRules)).toMatchSnapshot()
      })
    })

    describe('when the custom config attempts to amend a static value', () => {
      it('should not amend the static value', () => {
        const combinedRules = combineRules(
          defaultRules,
          {
            style: { solid: 'dashed' }
          },
          color
        )
        expect(border(combinedRules).style.solid['border-style']).toBe('solid')
      })
    })
  })
})
