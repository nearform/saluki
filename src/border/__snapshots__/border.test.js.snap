// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`border when custom config is provided when the custom config overrides a value should replace the default value 1`] = `
Object {
  "border": "solid 1px black",
  "bottom": Object {
    "dashed": Object {
      "border-bottom-style": "dashed",
    },
    "dotted": Object {
      "border-bottom-style": "dotted",
    },
    "extraThick": Object {
      "border-bottom-width": "8px",
    },
    "medium": Object {
      "border-bottom-width": "2px",
    },
    "none": Object {
      "border-bottom-style": "none",
    },
    "solid": Object {
      "border-bottom-style": "solid",
    },
    "thick": Object {
      "border-bottom-width": "45px",
    },
    "thin": Object {
      "border-bottom-width": "1px",
    },
  },
  "dashed": Object {
    "border-style": "dashed",
  },
  "dotted": Object {
    "border-style": "dotted",
  },
  "extraThick": Object {
    "border-width": "8px",
  },
  "left": Object {
    "dashed": Object {
      "border-left-style": "dashed",
    },
    "dotted": Object {
      "border-left-style": "dotted",
    },
    "extraThick": Object {
      "border-left-width": "8px",
    },
    "medium": Object {
      "border-left-width": "2px",
    },
    "none": Object {
      "border-left-style": "none",
    },
    "solid": Object {
      "border-left-style": "solid",
    },
    "thick": Object {
      "border-left-width": "45px",
    },
    "thin": Object {
      "border-left-width": "1px",
    },
  },
  "medium": Object {
    "border-width": "2px",
  },
  "none": Object {
    "border-style": "none",
  },
  "right": Object {
    "dashed": Object {
      "border-right-style": "dashed",
    },
    "dotted": Object {
      "border-right-style": "dotted",
    },
    "extraThick": Object {
      "border-right-width": "8px",
    },
    "medium": Object {
      "border-right-width": "2px",
    },
    "none": Object {
      "border-right-style": "none",
    },
    "solid": Object {
      "border-right-style": "solid",
    },
    "thick": Object {
      "border-right-width": "45px",
    },
    "thin": Object {
      "border-right-width": "1px",
    },
  },
  "solid": Object {
    "border-style": "solid",
  },
  "thick": Object {
    "border-width": "45px",
  },
  "thin": Object {
    "border-width": "1px",
  },
  "top": Object {
    "dashed": Object {
      "border-top-style": "dashed",
    },
    "dotted": Object {
      "border-top-style": "dotted",
    },
    "extraThick": Object {
      "border-top-width": "8px",
    },
    "medium": Object {
      "border-top-width": "2px",
    },
    "none": Object {
      "border-top-style": "none",
    },
    "solid": Object {
      "border-top-style": "solid",
    },
    "thick": Object {
      "border-top-width": "45px",
    },
    "thin": Object {
      "border-top-width": "1px",
    },
  },
}
`;

exports[`border when custom config is provided when the custom config provides a new value should add the value 1`] = `
Object {
  "border": "solid 1px black",
  "bottom": Object {
    "dashed": Object {
      "border-bottom-style": "dashed",
    },
    "dotted": Object {
      "border-bottom-style": "dotted",
    },
    "extraThick": Object {
      "border-bottom-width": "8px",
    },
    "medium": Object {
      "border-bottom-width": "2px",
    },
    "none": Object {
      "border-bottom-style": "none",
    },
    "solid": Object {
      "border-bottom-style": "solid",
    },
    "superThick": Object {
      "border-bottom-width": "100px",
    },
    "thick": Object {
      "border-bottom-width": "4px",
    },
    "thin": Object {
      "border-bottom-width": "1px",
    },
  },
  "dashed": Object {
    "border-style": "dashed",
  },
  "dotted": Object {
    "border-style": "dotted",
  },
  "extraThick": Object {
    "border-width": "8px",
  },
  "left": Object {
    "dashed": Object {
      "border-left-style": "dashed",
    },
    "dotted": Object {
      "border-left-style": "dotted",
    },
    "extraThick": Object {
      "border-left-width": "8px",
    },
    "medium": Object {
      "border-left-width": "2px",
    },
    "none": Object {
      "border-left-style": "none",
    },
    "solid": Object {
      "border-left-style": "solid",
    },
    "superThick": Object {
      "border-left-width": "100px",
    },
    "thick": Object {
      "border-left-width": "4px",
    },
    "thin": Object {
      "border-left-width": "1px",
    },
  },
  "medium": Object {
    "border-width": "2px",
  },
  "none": Object {
    "border-style": "none",
  },
  "right": Object {
    "dashed": Object {
      "border-right-style": "dashed",
    },
    "dotted": Object {
      "border-right-style": "dotted",
    },
    "extraThick": Object {
      "border-right-width": "8px",
    },
    "medium": Object {
      "border-right-width": "2px",
    },
    "none": Object {
      "border-right-style": "none",
    },
    "solid": Object {
      "border-right-style": "solid",
    },
    "superThick": Object {
      "border-right-width": "100px",
    },
    "thick": Object {
      "border-right-width": "4px",
    },
    "thin": Object {
      "border-right-width": "1px",
    },
  },
  "solid": Object {
    "border-style": "solid",
  },
  "superThick": Object {
    "border-width": "100px",
  },
  "thick": Object {
    "border-width": "4px",
  },
  "thin": Object {
    "border-width": "1px",
  },
  "top": Object {
    "dashed": Object {
      "border-top-style": "dashed",
    },
    "dotted": Object {
      "border-top-style": "dotted",
    },
    "extraThick": Object {
      "border-top-width": "8px",
    },
    "medium": Object {
      "border-top-width": "2px",
    },
    "none": Object {
      "border-top-style": "none",
    },
    "solid": Object {
      "border-top-style": "solid",
    },
    "superThick": Object {
      "border-top-width": "100px",
    },
    "thick": Object {
      "border-top-width": "4px",
    },
    "thin": Object {
      "border-top-width": "1px",
    },
  },
}
`;

exports[`border when no custom config is provided should return the default config 1`] = `
Object {
  "black": Object {
    "border-color": "black",
  },
  "blue": Object {
    "border-color": "blue",
  },
  "border": "solid 1px black",
  "bottom": Object {
    "black": Object {
      "border-bottom-color": "black",
    },
    "blue": Object {
      "border-bottom-color": "blue",
    },
    "dashed": Object {
      "border-bottom-style": "dashed",
    },
    "dotted": Object {
      "border-bottom-style": "dotted",
    },
    "extraThick": Object {
      "border-bottom-width": "8px",
    },
    "gray": Object {
      "border-bottom-color": "gray",
    },
    "green": Object {
      "border-bottom-color": "green",
    },
    "medium": Object {
      "border-bottom-width": "2px",
    },
    "none": Object {
      "border-bottom-style": "none",
    },
    "orange": Object {
      "border-bottom-color": "orange",
    },
    "purple": Object {
      "border-bottom-color": "purple",
    },
    "red": Object {
      "border-bottom-color": "red",
    },
    "solid": Object {
      "border-bottom-style": "solid",
    },
    "thick": Object {
      "border-bottom-width": "4px",
    },
    "thin": Object {
      "border-bottom-width": "1px",
    },
    "white": Object {
      "border-bottom-color": "white",
    },
    "yellow": Object {
      "border-bottom-color": "yellow",
    },
  },
  "dashed": Object {
    "border-style": "dashed",
  },
  "dotted": Object {
    "border-style": "dotted",
  },
  "extraThick": Object {
    "border-width": "8px",
  },
  "gray": Object {
    "border-color": "gray",
  },
  "green": Object {
    "border-color": "green",
  },
  "left": Object {
    "black": Object {
      "border-left-color": "black",
    },
    "blue": Object {
      "border-left-color": "blue",
    },
    "dashed": Object {
      "border-left-style": "dashed",
    },
    "dotted": Object {
      "border-left-style": "dotted",
    },
    "extraThick": Object {
      "border-left-width": "8px",
    },
    "gray": Object {
      "border-left-color": "gray",
    },
    "green": Object {
      "border-left-color": "green",
    },
    "medium": Object {
      "border-left-width": "2px",
    },
    "none": Object {
      "border-left-style": "none",
    },
    "orange": Object {
      "border-left-color": "orange",
    },
    "purple": Object {
      "border-left-color": "purple",
    },
    "red": Object {
      "border-left-color": "red",
    },
    "solid": Object {
      "border-left-style": "solid",
    },
    "thick": Object {
      "border-left-width": "4px",
    },
    "thin": Object {
      "border-left-width": "1px",
    },
    "white": Object {
      "border-left-color": "white",
    },
    "yellow": Object {
      "border-left-color": "yellow",
    },
  },
  "medium": Object {
    "border-width": "2px",
  },
  "none": Object {
    "border-style": "none",
  },
  "orange": Object {
    "border-color": "orange",
  },
  "purple": Object {
    "border-color": "purple",
  },
  "red": Object {
    "border-color": "red",
  },
  "right": Object {
    "black": Object {
      "border-right-color": "black",
    },
    "blue": Object {
      "border-right-color": "blue",
    },
    "dashed": Object {
      "border-right-style": "dashed",
    },
    "dotted": Object {
      "border-right-style": "dotted",
    },
    "extraThick": Object {
      "border-right-width": "8px",
    },
    "gray": Object {
      "border-right-color": "gray",
    },
    "green": Object {
      "border-right-color": "green",
    },
    "medium": Object {
      "border-right-width": "2px",
    },
    "none": Object {
      "border-right-style": "none",
    },
    "orange": Object {
      "border-right-color": "orange",
    },
    "purple": Object {
      "border-right-color": "purple",
    },
    "red": Object {
      "border-right-color": "red",
    },
    "solid": Object {
      "border-right-style": "solid",
    },
    "thick": Object {
      "border-right-width": "4px",
    },
    "thin": Object {
      "border-right-width": "1px",
    },
    "white": Object {
      "border-right-color": "white",
    },
    "yellow": Object {
      "border-right-color": "yellow",
    },
  },
  "solid": Object {
    "border-style": "solid",
  },
  "thick": Object {
    "border-width": "4px",
  },
  "thin": Object {
    "border-width": "1px",
  },
  "top": Object {
    "black": Object {
      "border-top-color": "black",
    },
    "blue": Object {
      "border-top-color": "blue",
    },
    "dashed": Object {
      "border-top-style": "dashed",
    },
    "dotted": Object {
      "border-top-style": "dotted",
    },
    "extraThick": Object {
      "border-top-width": "8px",
    },
    "gray": Object {
      "border-top-color": "gray",
    },
    "green": Object {
      "border-top-color": "green",
    },
    "medium": Object {
      "border-top-width": "2px",
    },
    "none": Object {
      "border-top-style": "none",
    },
    "orange": Object {
      "border-top-color": "orange",
    },
    "purple": Object {
      "border-top-color": "purple",
    },
    "red": Object {
      "border-top-color": "red",
    },
    "solid": Object {
      "border-top-style": "solid",
    },
    "thick": Object {
      "border-top-width": "4px",
    },
    "thin": Object {
      "border-top-width": "1px",
    },
    "white": Object {
      "border-top-color": "white",
    },
    "yellow": Object {
      "border-top-color": "yellow",
    },
  },
  "white": Object {
    "border-color": "white",
  },
  "yellow": Object {
    "border-color": "yellow",
  },
}
`;
